src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -67,24 +65,15 @@ internal partial class CodeFixService : ICodeFixService
             [ImportMany] IEnumerable<Lazy<CodeFixProvider, CodeChangeProviderMetadata>> fixers,
             [ImportMany] IEnumerable<Lazy<IConfigurationFixProvider, CodeChangeProviderMetadata>> configurationProviders)
         {
-            _errorLoggers = loggers;
             _diagnosticService = diagnosticAnalyzerService;
+            _errorLoggers = loggers.ToImmutableArray();
 
-            _lazyFixerToMetadataMap = new(() => fixers.Where(service => service.IsValueCreated).ToImmutableDictionary(service => service.Value, service => service.Metadata));
-            var fixersPerLanguageMap = fixers.ToPerLanguageMapWithMultipleLanguages();
-            var configurationProvidersPerLanguageMap = configurationProviders.ToPerLanguageMapWithMultipleLanguages();
-
-            _getWorkspaceFixersMap = workspace => GetFixerPerLanguageMap(fixersPerLanguageMap, workspace);
-            _configurationProvidersMap = GetConfigurationProvidersPerLanguageMap(configurationProvidersPerLanguageMap);
+            _fixers = fixers.ToImmutableArray();
+            _fixersPerLanguageMap = _fixers.ToPerLanguageMapWithMultipleLanguages();
 
-            // REVIEW: currently, fixer's priority is statically defined by the fixer itself. might considering making it more dynamic or configurable.
-            _getFixerPriorityMap = workspace => GetFixerPriorityPerLanguageMap(fixersPerLanguageMap, workspace);
+            _lazyFixerToMetadataMap = new(() => fixers.Where(service => service.IsValueCreated).ToImmutableDictionary(service => service.Value, service => service.Metadata));
```

Comment:
Out of curiosity, when this runs in practice are there any fixers already created or is this just initializing an empty map?

Reply:
(why this is stuffed in a Lazy of its own is terrifying...)

Reply:
@mavasani this is a bit scary.  this lazy is exposed through:

```c#
private ImmutableDictionary<CodeFixProvider, CodeChangeProviderMetadata> FixerToMetadataMap => _lazyFixerToMetadataMap.Value;
```

This means that we capture whatever the state of hte world is when this is computed.  Meaning if a service hasn't been created, then it's never in this map.  This seems fishy, esp in a multi-language world.  How do we know that we won't capture a snapshot of things in a bad state?

Reply:
attempting to fix that here: https://github.com/dotnet/roslyn/pull/59586